static/js/timerender.js

```diff
@@ -20,6 +22,37 @@ export function clear_for_testing() {
     next_timerender_id = 0;
 }
 
+// Exported for tests only.
+export function get_tz_with_UTC_offset(time) {
+    // Convert to number of hours ahead/behind UTC.
+    // The sign of getTimezoneOffset() is reversed wrt
+    // the conventional meaning of UTC+n / UTC-n
+    const tz_offset = -time.getTimezoneOffset();
+    const tz_offset_hrs = Math.floor(tz_offset / 60);
+    const tz_offset_mins = tz_offset % 60;
+
+    let timezone = new Intl.DateTimeFormat(undefined, {timeZoneName: "short"})
+        .formatToParts(time)
+        .find(({type}) => type === "timeZoneName").value;
+
+    if (timezone === "UTC") {
+        return "UTC";
+    }
+
+    // In some rare cases where user's locale and timeZone doesn't suit,
+    // for eg. en-US with Indian timeZone gives GMT+5:30. We want to
+    // avoid showing something like: 27/6/21 at 12:00 GMT+x:y (UTC+N).
+    timezone = /GMT[+-][\d:]*/.test(timezone) ? "" : timezone;
+
+    const tz_offset_sign = tz_offset >= 0 ? "+" : ""; // - not needed if already negative.
+    const tz_UTC_offset = `(UTC${tz_offset_sign}${tz_offset_hrs}:${tz_offset_mins})`;
```

Comment:
There are still many bugs here. UTC-07:00 is incorrectly displayed as “UTC-7:0”, and UTC-03:30 is incorrectly displayed as “UTC-4:-30”(!).

We should let `Intl.DateTimeFormat` or `date-fns` compute this for us, and not try to reimplement any timezone-related algorithms ourselves. (If that means we need to slightly change the format, that’s fine.)

Reply:
Now changed to use `format` of `date-fns`. There wasn't a need to change the format. We can still show something like `IST (UTC+05:30)`.